runtime/connections.go

```diff
@@ -340,37 +329,14 @@ func (c *ConnectorConfig) Resolve() map[string]any {
 	if n == 0 {
 		return nil
 	}
-
 	cfg := make(map[string]any, n)
-	for k, v := range c.Project {
-		cfg[strings.ToLower(k)] = v
-	}
-	for k, v := range c.Env {
-		cfg[strings.ToLower(k)] = v
-	}
-	for k, v := range c.Preset {
-		cfg[strings.ToLower(k)] = v
-	}
-	return cfg
-}
-
-// ResolveString is similar to Resolve, but it returns a map of strings.
-func (c *ConnectorConfig) ResolveStrings() map[string]string {
-	n := len(c.Preset) + len(c.Project) + len(c.Env)
-	if n == 0 {
-		return nil
-	}
-
-	cfg := make(map[string]string, n)
-	for k, v := range c.Project {
-		cfg[strings.ToLower(k)] = v
+	for k, v := range lowerKeys(c.Project) {
+		cfg[k] = v
```

Comment:
This modifies `c.Project` in-place, doesn't it? Seems weird to have side effects in this place. 

Maybe just keep the old code?
```suggestion
	for k, v := range c.Project {
		cfg[strings.ToLower(k)] = v
```